s3pd/__init__.py

```diff
@@ -127,10 +160,11 @@ def s3pd(url, processes=8, chunksize=67108864, destination=None, func=None,
     assert chunksize % mmap.ALLOCATIONGRANULARITY == 0
 
     parsed_url = urlparse(url)
-    scheme = parsed_url.scheme
     bucket = parsed_url.netloc
     key = parsed_url.path[1:]
 
+    bucket, key = resolve_link(bucket, key, signed)
+
     client = create_client(signed)
     filesize = get_filesize(client, bucket, key)
```

Comment:
Minor: As mentioned before, could the client reused in the recursion, and the function returns (bucket, key, filesize)?

Reply:
Yes, re-using the client looks like a must!

Reply:
Fixed in 09a5034